docs/layouts/shortcodes/generated/dynamic_section.html

```diff
@@ -8,6 +8,24 @@
         </tr>
     </thead>
     <tbody>
+        <tr>
+            <td><h5>kubernetes.operator.checkpoint.cleanup.enabled</h5></td>
+            <td style="word-wrap: break-word;">true</td>
+            <td>Boolean</td>
+            <td>Whether to enable clean up of checkpoint FlinkStateSnapshot resources created by the operator automatically. This will only remove the FlinkStateSnapshot CR, and will not remove any data on the filesystem.</td>
+        </tr>
+        <tr>
+            <td><h5>kubernetes.operator.checkpoint.history.max.age</h5></td>
+            <td style="word-wrap: break-word;">1 d</td>
+            <td>Duration</td>
+            <td>Maximum age for checkpoint FlinkStateSnapshot resources to retain. Due to lazy clean-up, the most recent checkpoint may live longer than the max age.</td>
+        </tr>
+        <tr>
+            <td><h5>kubernetes.operator.checkpoint.history.max.count</h5></td>
+            <td style="word-wrap: break-word;">10</td>
+            <td>Integer</td>
+            <td>Maximum number of checkpoint FlinkStateSnapshot resources to retain.</td>
+        </tr>
```

Comment:
I wonder if we could get rid of these 3 configs by:
 1. Always enabling cleanup (it doesn't actually delete any data anyways)
 2. For max count use the configured execution.checkpointing.num-retained (and get rid of the age based deletion for checkpoints)
 
 I think this will cover most use-cases and if we really need another cleanup policy for some reason then we can re-introduce this in the future maybe?

Reply:
That sounds good. I wasn't sure how fine-grained these config options should be, but if you think that we don't need these, I am happy to make this PR more simple and reduce the amount of config options. 

It will be very easy to add these config options back in the future if we deem them necessary anyways.

Reply:
Sounds good :) always better to start with simple / less configs for new features